my-code/significa/src/pages/Movie.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState, useEffect } from "react";
+import { useParams, Link } from "@reach/router";
+
+import Layout from "../components/layout";
+import { ButtonFavYes, ButtonFavNo, ButtonArrow } from "../components/Buttons";
+import { LabelLogo, LabelRated } from "../components/Labels";
+import logo from "../assets/logos/logo-icon.svg";
+import "./Movie.css";
+
+const API_KEY = `${process.env.REACT_APP_API_KEY}`;
+
+const Movie = () => {
+  /* define movie Id from passed arguments */
+  const params = useParams();
+
+  /* set variables that use state in the api call */
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [movie, setMovie] = useState([]);
+
+  /* making favourites persistent (using localStorage) */
+  let favourites = localStorage.getItem("whatsInFavourites");
+
+  const [favourite, setFavourite] = useState(() => {
```

Comment:
This isn't necessarily wrong, but the initial state of the state should not take care of this kind of logic but receives a value, not more than that. My suggestion would be:
- Create a util file to manage the localstorage, in order to avoid repeating code;
- There you'll be able to create abstraction around the localstorage and make it easier to work;
- So `useState` would receive a single value from this util:
```js
const { isItemFavorite } = useFavorite() // abstraction over localstorage
const [favourite, setFavourite] = useState(isItemFavorite(movieId))
```

Does it make sense?